### ISSUE TITLE
Wrong encoding of '/$$$%'

### ISSUE THREAD

**Issue by danielkasza** [rejectedsoftware/vibe.d#1318]
Hello,

I am not 100% sure if this is an issue or I am doing something wrong, but I could not find any reference to this problem.

I have a very simple vibe.d project:

````
import vibe.appmain;
import vibe.d;
import std.stdio;

void handleRequest(scope HTTPServerRequest req, scope HTTPServerResponse res) {
	writeln(req.path);
	res.writeBody(req.path, "text/plain");
}

shared static this() {
	auto settings = new HTTPServerSettings;
	settings.port = 9000;
	settings.bindAddresses = ["0.0.0.0"];
	
	listenHTTP(settings, &handleRequest);
}
````

````
{
	"name" : "firstvibe",
	"description" : "Hello World - A minimal DUB bundle.",
	"dependencies" : {
		"vibe-d": "~>0.7.26-rc.3"
	},
	"versions": ["VibeDefaultMain"]
}
````

This works fine for most paths, but if I try to open `http://localhost:9000/$$$%#$@%`, the process dies with this exception:

````
Task terminated with unhandled exception: Wrong encoding of '/$$$%'
````

Is there any way to properly handle this?

**Comment by s-ludwig** [comment]
I've committed a fix that emits an `Exception` instead of an `AssertError` as expected. This somehow got through in the PR review that was actually supposed to get this fixed. It will now result in an error message displayed in the browser (to avoid the error completely, replace `"%"` with  `"%25"` to get valid URL encoding).

The fix will be in the upcoming 0.7.26 version.

**Comment by danielkasza** [comment]
Thank you!
